From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pekka Vuorela <pekka.vuorela@jolla.com>
Date: Wed, 6 Apr 2022 17:56:07 +0300
Subject: [PATCH] Install unit test files

Before switch to cmake on commit 7471153ce the unit test files were
installed, and with cmake there are some parts of it done.
Reintroduce the installation as these are needed if the build
environment is not, and perhap cannot be, used for testing.

Also cope on the qml unit test for the example plugin not availabe
as it's built optionally.
---
 CMakeLists.txt                                       | 12 ++++++++++--
 .../ut_minputmethodquickplugin.cpp                   |  5 ++++-
 2 files changed, 14 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a29ea4f..a1cebdd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -511,6 +511,7 @@ if(enable-tests)
     set(DUMMY_PLUGINS dummyimplugin dummyimplugin2 dummyimplugin3 dummyplugin)
 
     set_property(TARGET ${DUMMY_PLUGINS} PROPERTY LIBRARY_OUTPUT_DIRECTORY ${TEST_PLUGINS_DIR})
+    install(TARGETS ${DUMMY_PLUGINS} DESTINATION ${CMAKE_INSTALL_LIBDIR}/maliit-framework-tests/plugins)
 
     add_library(test-utils STATIC
             tests/utils/core-utils.cpp
@@ -520,8 +521,8 @@ if(enable-tests)
     target_link_libraries(test-utils PUBLIC Qt5::Core Qt5::Gui Qt5::Test maliit-connection)
     target_include_directories(test-utils INTERFACE tests/utils)
     target_compile_definitions(test-utils PUBLIC
-            -DMALIIT_TEST_PLUGINS_DIR="${CMAKE_INSTALL_PREFIX}/lib/maliit-framework-tests/plugins"
-            -DMALIIT_TEST_DATA_PATH="${CMAKE_INSTALL_PREFIX}/lib/maliit-framework-tests"
+            -DMALIIT_TEST_PLUGINS_DIR="${CMAKE_INSTALL_FULL_LIBDIR}/maliit-framework-tests/plugins"
+            -DMALIIT_TEST_DATA_PATH="${CMAKE_INSTALL_FULL_LIBDIR}/maliit-framework-tests"
             -DIN_TREE_TEST_PLUGIN_DIR="${CMAKE_SOURCE_DIR}/examples/plugins")
 
     add_library(test-stubs STATIC
@@ -540,14 +541,21 @@ if(enable-tests)
         list(REMOVE_AT _libs 0)
         target_link_libraries(${name} test-utils maliit-plugins ${_libs})
         add_test(${name} ${name})
+        install(TARGETS ${name} DESTINATION ${CMAKE_INSTALL_LIBDIR}/maliit-framework-tests/${name})
 
         set(test_targets ${test_targets} ${name} PARENT_SCOPE)
     endfunction()
 
     create_test(sanitychecks)
     create_test(ut_mattributeextensionmanager)
+    install(DIRECTORY tests/ut_mattributeextensionmanager/
+           DESTINATION ${CMAKE_INSTALL_LIBDIR}/maliit-framework-tests/ut_mattributeextensionmanager
+           FILES_MATCHING PATTERN "*.xml")
     create_test(ut_mimonscreenplugins)
     create_test(ut_mimpluginmanager ${DUMMY_PLUGINS})
+    install(DIRECTORY tests/ut_mimpluginmanager/
+           DESTINATION ${CMAKE_INSTALL_LIBDIR}/maliit-framework-tests/ut_mimpluginmanager
+           FILES_MATCHING PATTERN "*.xml")
     create_test(ut_mimpluginmanagerconfig)
     create_test(ut_mimserveroptions)
     create_test(ut_mimsettings)
diff --git a/tests/ut_minputmethodquickplugin/ut_minputmethodquickplugin.cpp b/tests/ut_minputmethodquickplugin/ut_minputmethodquickplugin.cpp
index 33ec902..4b7c4f2 100644
--- a/tests/ut_minputmethodquickplugin/ut_minputmethodquickplugin.cpp
+++ b/tests/ut_minputmethodquickplugin/ut_minputmethodquickplugin.cpp
@@ -64,7 +64,10 @@ void Ut_MInputMethodQuickPlugin::testQmlSetup()
                 QDir(IN_TREE_TEST_PLUGIN_DIR"/qml") : QDir(MALIIT_TEST_PLUGINS_DIR"/examples/qml");
     const QString pluginPath = pluginDir.absoluteFilePath(testPluginPath);
     const QString pluginId = QFileInfo(testPluginPath).baseName();
-    QVERIFY(pluginDir.exists(pluginPath));
+
+    if (!pluginDir.exists(pluginPath)) {
+        QSKIP("QML Example plugin not available");
+    }
 
     QObject *pluginInstance = 0;
     Maliit::Plugins::InputMethodPlugin *plugin = 0;
-- 
2.35.1

